_source/_posts/2021-07-14-flask-scim-server.md

```diff
@@ -0,0 +1,519 @@
+---
+layout: blog_post
+title: "Okta, SCIM and Flask"
+author: cale-switzer
+by: advocate
+communities: [python]
+description: "Using a Flask SCIM server, learn more about the how & why behind how Okta implements the provisioning protocol."
+tags: [scim, python, flask]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+
+At a very high level, the SCIM (System for Cross-Identity Management) protocol exists to provision and sync users and groups across various, independent systems. There is some room for nuance in the [SCIM spec](https://datatracker.ietf.org/doc/html/rfc7644). In this post we will take a look at some basic SCIM operations and what the flow looks like under the hood between Okta and a SCIM server built in Flask using PostgreSQL as our app database. Even if you're not a Python dev, hopefully you can take away some of the concepts in this guide for your own SCIM implementation.
+
+### What you'll need
+
+- [virtualenv](https://docs.python.org/3/library/venv.html)
+- [PostgreSQL](https://www.postgresql.org/)
+- [ngrok](https://ngrok.com/)
+
+### Setting up the Flask server and Postgres DB 
+{% img blog/flask-scim-server/flask_postgres.png alt:"PostgreSQL and Flask Logos Image" width:"500" %}
+
+1.  Clone the repo found [here](https://github.com/Tennyx/flask-scim-server), open terminal and `cd` into the project root.
+2.  Create a new virtualenv in the root folder with `virtualenv env`
+3.  Run the virtual environment with `source env/bin/activate`
+4.  Install necessary Python packages with ```pip install -r requirements.txt```
+5.  Create a new Postgres database named `scim`. Enter the `psql` shell by opening a new terminal tab and typing `psql postgres`. Create the DB with `CREATE DATABASE scim;` (Run `\l` to double-check the database has been created)
```

Comment:
I suggest using the long form of the commands. For example `\list` instead of `\l`. I know that the long forms take longer to type, but the long form commands make it clear to new readers what exactly they will be doing.